middleware/kubernetes/kubernetes.go

```diff
@@ -270,32 +314,66 @@ func (k *Kubernetes) getRecordsForServiceItems(serviceItems []*api.Service, zone
 }
 
 // Get performs the call to the Kubernetes http API.
-func (k *Kubernetes) Get(namespace string, nsWildcard bool, servicename string, serviceWildcard bool, typeName string) ([]*api.Service, error) {
+func (k *Kubernetes) Get(namespace string, nsWildcard bool, servicename string, serviceWildcard bool, endpointname string, port string, portWildcard bool, protocol string, protocolWildcard bool, typeName string) (services []service, err error) {
 	switch {
 	case typeName == "pod":
-		return nil, fmt.Errorf("pod not implemented")
+		return nil, fmt.Errorf("%v not implemented", typeName)
 	default:
-		return k.getServices(namespace, nsWildcard, servicename, serviceWildcard)
+		return k.getServices(namespace, nsWildcard, servicename, serviceWildcard, endpointname, port, portWildcard, protocol, protocolWildcard)
 	}
 }
 
-func (k *Kubernetes) getServices(namespace string, nsWildcard bool, servicename string, serviceWildcard bool) ([]*api.Service, error) {
+func (k *Kubernetes) getServices(namespace string, nsWildcard bool, servicename string, serviceWildcard bool, endpointname string, port string, portWildcard bool, protocol string, protocolWildcard bool) ([]service, error) {
```

Comment:
Wow, this 9 parameters :) We def. need some structure that holds these, maybe we can drop all those wildcard booleans into something like that?